packages/parameters/src/ssm/SSMProvider.ts

```diff
@@ -162,19 +161,19 @@ class SSMProvider extends BaseProvider {
     options?: SSMGetMultipleOptionsInterface
   ): Promise<Record<string, string | undefined>> {
     const sdkOptions: GetParametersByPathCommandInput = {
+      ...(options?.sdkOptions || {}),
       Path: path,
     };
     const paginationOptions: PaginationConfiguration = {
       client: this.client
     };
     if (options) {
-      if (options.hasOwnProperty('decrypt')) sdkOptions.WithDecryption = options.decrypt;
-      if (options.hasOwnProperty('recursive')) sdkOptions.Recursive = options.recursive;
-      if (options.hasOwnProperty('sdkOptions')) {
-        Object.assign(sdkOptions, options.sdkOptions);
-        if (sdkOptions.MaxResults) {
-          paginationOptions.pageSize = sdkOptions.MaxResults;
-        }
+      if (options.hasOwnProperty('decrypt'))
+        sdkOptions.WithDecryption = options.decrypt;
+      if (options.hasOwnProperty('recursive'))
+        sdkOptions.Recursive = options.recursive;
+      if (sdkOptions.MaxResults) {
+        paginationOptions.pageSize = sdkOptions.MaxResults;
```

Comment:
Similar to previous typing comment we can assign properties of they are present in options:
```ts
    sdkOptions.WithDecryption = options?.decrypt ? options.decrypt : undefined;
    sdkOptions.Recursive = options?.recursive ? options.recursive : undefined;
    paginationOptions.pageSize = sdkOptions?.MaxResults ? sdkOptions.MaxResults : undefined;
```

Reply:
So I initially thought about this, but I wasn't sure on whether setting these values to `undefined` would overwrite the default values/behavior of the SDK, so I opted for assigning only when the user actually provides a value, otherwise leave them as they are

Reply:
I'm going to follow your suggestion.

I have made a quick test with a middleware that logs right before sending the request and thankfully the sdk is smart enough to remove any key that has value `undefined` before sending the request, so it's safe to do like you suggest.